lib/ex_doc/formatter/html.ex

```diff
@@ -23,14 +23,20 @@ defmodule ExDoc.Formatter.HTML do
     exceptions = filter_list(:exceptions, all)
     protocols  = filter_list(:protocols, all)
 
-    has_readme = config.readme && generate_readme(output, module_nodes, config, modules, exceptions, protocols)
+    if config.logo do
+      logo_path = process_logo_metadata(config)
+    else
+      logo_path = nil
+    end
+
+    has_readme = config.readme && generate_readme(output, module_nodes, config, modules, exceptions, protocols, logo_path)
     generate_index(output, config)
-    generate_overview(modules, exceptions, protocols, output, config, has_readme)
-    generate_not_found(modules, exceptions, protocols, output, config, has_readme)
+    generate_overview(modules, exceptions, protocols, output, config, has_readme, logo_path)
```

Comment:
Instead of passing logo_path around, create a `logo_path(config)` function that returns the proper path. This way, we don't  need to pass even more data to the templates.